src/include/sof/platform.h

```diff
@@ -54,6 +54,24 @@ int platform_context_save(struct sof *sof);
 
 /** @}*/
 
+#define HOST_TO_LOCAL 0
+#define LOCAL_TO_HOST 1
+
+static inline void convert_addr(int type, uint32_t *addr)
+{
+#ifdef CONFIG_IMX8ULP
+	switch (type) {
+	case HOST_TO_LOCAL:
+		*addr = SDRAM0_BASE + (*addr - MEM_RESERVED);
+		break;
+	case LOCAL_TO_HOST:
+		*addr = MEM_RESERVED + (*addr - SDRAM0_BASE);
+		break;
+	};
+#endif
+}
+#else
```

Comment:
this `#else` doesn't seem to be needed

Reply:
Sure. Will remove this.

Reply:
We left this out of this PR because it is sensible and it needs careful review from more people.